cli/multisig.go

```diff
@@ -106,69 +100,21 @@ var msigCreateCmd = &cli.Command{
 			return err
 		}
 
+		intVal := types.BigInt(filval)
+
 		required := cctx.Int64("required")
 		if required == 0 {
 			required = int64(len(addrs))
 		}
 
-		// Set up constructor parameters for multisig
-		msigParams := &samsig.ConstructorParams{
-			Signers:               addrs,
-			NumApprovalsThreshold: required,
-		}
-
-		enc, err := actors.SerializeParams(msigParams)
-		if err != nil {
-			return err
-		}
-
-		// new actors are created by invoking 'exec' on the init actor with the constructor params
-		execParams := &init_.ExecParams{
-			CodeCID:           builtin.MultisigActorCodeID,
-			ConstructorParams: enc,
-		}
-
-		enc, err = actors.SerializeParams(execParams)
-		if err != nil {
-			return err
-		}
-
-		// now we create the message to send this with
-		msg := types.Message{
-			To:       builtin.InitActorAddr,
-			From:     sendAddr,
-			Method:   builtin.MethodsInit.Exec,
-			Params:   enc,
-			GasPrice: types.NewInt(1),
-			GasLimit: 1000000,
-			Value:    types.BigInt(filval),
-		}
-
-		// send the message out to the network
-		smsg, err := api.MpoolPushMessage(ctx, &msg)
-		if err != nil {
-			return err
-		}
+		gp := types.NewInt(1)
 
-		// wait for it to get mined into a block
-		wait, err := api.StateWaitMsg(ctx, smsg.Cid())
+		addressPair, err := api.MsigCreate(ctx, required, addrs, intVal, sendAddr, gp)
 		if err != nil {
 			return err
 		}
 
-		// check it executed successfully
-		if wait.Receipt.ExitCode != 0 {
-			fmt.Println("actor creation failed!")
-			return err
-		}
-
-		// get address of newly created miner
-
-		var execreturn init_.ExecReturn
-		if err := execreturn.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
-			return err
-		}
-		fmt.Println("Created new multisig: ", execreturn.IDAddress, execreturn.RobustAddress)
+		fmt.Println("Created new multisig: ", addressPair.IDAddress, addressPair.RobustAddress)
```

Comment:
Somebody more familiar with Lotus can speak to this but I'm not sure if exposing the robust address is a good idea. (This is the first I've heard of it, honestly!) If nothing else, they should be marked as ID and robust addresses.

Reply:
To my knowledge there is no reason to keep this private. Also, this merely mimics the output of the cli command. `lotus msig create` currently returns both the ID and Robust addresses.

Reply:
Fair enough. I am curious about `RobustAddress`, though. I haven't seen that before. Can you give some examples from your testing? Are they `t0` addresses or something else?

Reply:
Yea sure! Take a look at the PR description above under `Output` for `MsigCreate`. Although I'm not sure if it is always guaranteed to be the case, in my example it is a t2 address (Actor Address) I'm assuming Robust just means non-ID address but @whyrusleeping or @magik6k can speak to that.

Reply:
Thanks. Sounds like it's just another name for actor addresses but somebody else can chime in and confirm. :)

Reply:
Yeah, the robust address is stable across chain reorgs. The ID address may change if the block including your creation message is reorged